src/fq.rs

```diff
@@ -65,6 +65,20 @@ fn adc(a: u64, b: u64, carry: &mut u128) -> u64 {
     *carry as u64
 }
 
+/// Compute a + b + carry, returning the result the new carry over.
+#[inline(always)]
+fn adc2(a: u64, b: u64, carry: u64) -> (u64, u64) {
+    let adc = u128::from(a) + u128::from(b) + u128::from(carry);
+    (adc as u64, (adc >> 64) as u64)
+}
+
+/// Compute a + b, returning the result and the carry over.
+#[inline(always)]
+fn overflowing_add(a: u64, b: u64) -> (u64, u64) {
+    let (sum, overflow) = a.overflowing_add(b);
```

Comment:
I think we should use u128 and split, the compiler will do the same thing and it's consistent with the other functions. I suspect actually that running `adc` (or `adc2` in your current PR) with `0` as an operand will be automatically optimized into what you've done.

Reply:
Yeah in doing some of this work, I began to suspect that the compiler was optimizing automatically the situation where we are adding 0.